jackson/src/main/java/com/baeldung/jackson/enums/Distance.java

```diff
@@ -1,20 +1,29 @@
 package com.baeldung.jackson.enums;
 
 import com.baeldung.jackson.serialization.DistanceSerializer;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 
 /**
  * Use  @JsonFormat to handle representation of Enum as JSON (available since Jackson 2.1.2)
  * Use @JsonSerialize to configure a custom Jackson serializer
  */
-// @JsonFormat(shape = JsonFormat.Shape.OBJECT)
-@JsonSerialize(using = DistanceSerializer.class)
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+//@JsonSerialize(using = DistanceSerializer.class)
 public enum Distance {
-    KILOMETER("km", 1000), MILE("miles", 1609.34), METER("meters", 1), INCH("inches", 0.0254), CENTIMETER("cm", 0.01), MILLIMETER("mm", 0.001);
+    @JsonProperty("distance-in-km")
+    KILOMETER("km", 1000), 
+    @JsonProperty("distance-in-miles")
+    MILE("miles", 1609.34), METER("meters", 1), INCH("inches", 0.0254), CENTIMETER("cm", 0.01), 
```

Comment:
let's continue the JsonProperty annotation for all the values, not just the ones we show in the article